tensorflow/core/kernels/cwise_ops.h

```diff
@@ -55,7 +55,7 @@ struct scalar_arg_op<std::complex<double>> {
 template <typename T>
 struct scalar_asinh_op {
   EIGEN_EMPTY_STRUCT_CTOR(scalar_asinh_op)
-  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const T operator()(const T& a) const {
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE T operator()(const T& a) const {
```

Comment:
I think this one is still required though (if T is a class or a pointer)

Reply:
Even if `T` is a pointer of type `U *`, then `const T` is of type `U * const`, not `const U *`, so it is still meaningless.

Reply:
If `T` is of type `U *`, `const T` will be of type `U * const`, not `const U *`. So it is still meaningless.